pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -72,6 +74,42 @@ func (h *AnnotationStateHistorian) RecordState(ctx context.Context, rule *ngmode
 	}
 }
 
+func buildAnnotationTextAndData(rule *ngmodels.AlertRule, currentState *state.State) (string, *simplejson.Json) {
+	jsonData := simplejson.New()
+	var value string
+
+	switch currentState.State {
+	case eval.Error:
+		if currentState.Error == nil {
+			jsonData.Set("error", nil)
+		} else {
+			jsonData.Set("error", currentState.Error.Error())
+		}
+		value = "Error"
+	case eval.NoData:
+		jsonData.Set("noData", true)
+		value = "No data"
+	default:
+		keys := make([]string, 0, len(currentState.Values))
+		for k := range currentState.Values {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		values := map[string]float64{}
```

Comment:
I don't think we need `values` here, we can just pass `currentState.Values` to `simplejson`?

Reply:
You are totally right. Thanks!